docs/http-api/openapi31/authoritative-api-openapi.yaml

```diff
@@ -0,0 +1,1311 @@
+          description: 'The name of this setting (e.g. ‘webserver-port’)'
+        value:
+          type: string
+          description: 'The value of setting name'
+
+    SimpleStatisticItem:
+      title: SimpleStatisticItem
+      type: object
+      additionalProperties: False
+      properties:
+        name:
+            type: string
+            description: 'Item name'
+        value:
+            type: string
+            description: 'Item value'
+
+    StatisticItem:
+      title: StatisticItem
+      type: object
+      additionalProperties: False
+      properties:
+        name:
+          type: string
+          description: 'Item name'
+        type:
+          type: string
+          description: 'set to "StatisticItem"'
+        value:
+          type: string
+          description: 'Item value'
+
+    MapStatisticItem:
+      title: MapStatisticItem
+      type: object
+      additionalProperties: False
+      properties:
+        name:
+          type: string
+          description: 'Item name'
+        type:
+          type: string
+          description: 'Set to "MapStatisticItem"'
+        value:
+          type: array
+          description: 'Named values'
+          items:
+            $ref: '#/components/schemas/SimpleStatisticItem'
+
+    RingStatisticItem:
+      title: RingStatisticItem
+      type: object
+      additionalProperties: False
+      properties:
+        name:
+          type: string
+          description: 'Item name'
+        type:
+          type: string
+          description: 'Set to "RingStatisticItem"'
+        size:
+          type: integer
+          description: 'Ring size'
+        value:
+          type: array
+          description: 'Named values'
+          items:
+            $ref: '#/components/schemas/SimpleStatisticItem'
+
+    SearchResultZone:
+      title: SearchResultZone
+      type: object
+      additionalProperties: False
+      properties:
+        name:
+          type: string
+        object_type:
+          type: string
+          description: 'set to "zone"'
+          const: zone
+        zone_id:
+          type: string
+
+    SearchResultRecord:
+      title: SearchResultRecord
+      type: object
+      additionalProperties: False
+      properties:
+        content:
+          type: string
+        disabled:
+          type: boolean
+        name:
+          type: string
+        object_type:
+          type: string
+          description: 'set to "record"'
+          const: record
+        zone_id:
+          type: string
+        zone:
+          type: string
+        type:
+          type: string
+        ttl:
+          type: integer
+
+    SearchResultComment:
+      title: SearchResultComment
+      type: object
+      additionalProperties: False
+      properties:
+        content:
+          type: string
+        name:
+          type: string
+        object_type:
+          type: string
+          description: 'set to "comment"'
+          const: comment
```

Comment:
The API returns another `type` (string) property after this. It contains the type of the Record that the comment is attached to (e.g. `CNAME`).

Reply:
The attribute is missing in the swagger doc as well. 
The openapi31 dd just makes use of additionalProperties=false on all objects to fault on unknown attributes.
Unit testing would help with these defects.